core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala

```diff
@@ -2845,7 +2844,178 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
     } finally {
       consumerSet.foreach(consumer => Utils.closeQuietly(consumer, "consumer"))
-      Utils.closeQuietly(producer, "producer")
+      Utils.closeQuietly(client, "adminClient")
+    }
+  }
+
+  @Test
+  def testDeleteShareGroupOffsets(): Unit = {
+    val config = createConfig
+    client = Admin.create(config)
+    val testTopicName = "test_topic"
+    val testGroupId = "test_group_id"
+    val testClientId = "test_client_id"
+    val fakeGroupId = "fake_group_id"
+    val fakeTopicName = "foo"
+
+    try {
+      prepareTopics(List(testTopicName), 1)
+      prepareRecords(testTopicName)
+
+      val newShareConsumerConfig = new Properties(consumerConfig)
+      newShareConsumerConfig.setProperty(ConsumerConfig.GROUP_ID_CONFIG, testGroupId)
+      newShareConsumerConfig.setProperty(ConsumerConfig.CLIENT_ID_CONFIG, testClientId)
+
+      Using.resource(createShareConsumer(configOverrides = newShareConsumerConfig)) { consumer =>
+        consumer.subscribe(util.List.of(testTopicName))
+        consumer.poll(JDuration.ofMillis(DEFAULT_MAX_WAIT_MS))
+        consumer.commitSync()
+
+        // listGroups is used to list share groups
+        // Test that we can list the new group.
+        TestUtils.waitUntilTrue(() => {
+          client.listGroups.all.get.stream().filter(group =>
+            group.groupId == testGroupId &&
+              group.groupState.get == GroupState.STABLE).count() == 1
+        }, s"Expected to be able to list $testGroupId")
+
+        // Test offset deletion while consuming
+        val offsetDeleteResult = client.deleteShareGroupOffsets(testGroupId, util.Set.of(testTopicName, fakeTopicName))
+
+        // Deleting the offset with real group ID should get GroupNotEmptyException
+        assertFutureThrows(classOf[GroupNotEmptyException], offsetDeleteResult.all())
+        assertFutureThrows(classOf[GroupNotEmptyException], offsetDeleteResult.topicResult(testTopicName))
+        assertFutureThrows(classOf[GroupNotEmptyException], offsetDeleteResult.topicResult(fakeTopicName))
+
+        // Test the fake group ID
+        val fakeDeleteResult = client.deleteShareGroupOffsets(fakeGroupId, util.Set.of(testTopicName, fakeTopicName))
+
+        assertFutureThrows(classOf[GroupIdNotFoundException], fakeDeleteResult.all())
+        assertFutureThrows(classOf[GroupIdNotFoundException], fakeDeleteResult.topicResult(testTopicName))
+        assertFutureThrows(classOf[GroupIdNotFoundException], fakeDeleteResult.topicResult(fakeTopicName))
+      }
+
+      // Test offset deletion when group is empty
+      val offsetDeleteResult = client.deleteShareGroupOffsets(testGroupId, util.Set.of(testTopicName, fakeTopicName))
+
+      assertFutureThrows(classOf[UnknownTopicOrPartitionException], offsetDeleteResult.all())
+      assertNull(offsetDeleteResult.topicResult(testTopicName).get())
+      assertFutureThrows(classOf[UnknownTopicOrPartitionException], offsetDeleteResult.topicResult(fakeTopicName))
+    } finally {
```

Comment:
We should be able to see whether offsets were deleted using `listShareGroupOffsets`.